### ISSUE TITLE
Test: extend synthetic-grammar test to full CLI pipeline (end-to-end generation integration)

### ISSUE THREAD

**Issue by mmcdermott** [mmcdermott/MEDS_EIC_AR#105]
## Summary

Extend the synthetic-grammar integration test (landed with #93 in `tests/test_pattern_generation.py`) to run through the full `MEICAR_pretrain` → `MEICAR_generate_trajectories` CLI pipeline instead of driving `Model.generate` in-process, and validate the output parquet files using the existing `GrammarFSM`. This gives us a true end-to-end correctness test for the generation pipeline — something we don't have today — with a natural home for stressing PR #103's N-trajectories interleaving at larger N and for covering CLI error paths we currently miss.

## Motivation

We currently have two related tests that each cover half of what we want:

1. **`test_pattern_generation.py`** — trains a tiny model on a synthetic grammar in-process and calls `Model.generate` directly. Proves **generation is content-correct** (grammar-valid continuations) but bypasses the whole CLI / Lightning / parquet-writing pipeline.
2. **`test_generate_trajectories.py::test_generate_trajectories_runs`** — drives the `MEICAR_generate_trajectories` CLI via subprocess on a tiny random-init pretrained model. Proves **the plumbing is wired** (files exist, subjects line up, trajectories are distinct) but can't validate content because the model hasn't learned anything.

Neither test alone can catch a class of bugs that lives in the interaction between the two — e.g., a CLI-layer regression that accidentally reorders rows between the interleaved predict pass and the per-trajectory parquet write, or a format_trajectories bug that shuffles subject metadata. PR #103's demux logic is exactly the kind of code that benefits from this test.

### A. Grammar-shaped MEDS dataset fixture

Build a minimal raw-MEDS dataset from the existing grammar. Each grammar sequence becomes one subject's timeline; each token becomes one MEDS event. The code-metadata parquet registers each grammar token as a distinct MEDS code, including `TIMELINE//END` for the dummy EOS.

```
subject 0: (time_0, A[0]) (time_1, A[1]) (time_2, A[2]) (time_3, A[3]) (time_4, SEP) (time_5, B[0]) ...
subject 1: (time_0, C[0]) (time_1, C[1]) ...
```

Timestamps can be arbitrary monotonic increments; the model only consumes codes.

### B. Session-scoped fixtures

- `grammar_preprocessed_dataset`: runs `MEICAR_process_data` on the raw grammar MEDS dir, producing a tensorized cohort dir.
- `grammar_pretrained_model`: runs `MEICAR_pretrain` on that cohort with a tiny model config (2 layers, hidden 32, max_seq_len 16) for a short budget. Aim: <60s on CI.
- `grammar_generated_trajectories`: runs `MEICAR_generate_trajectories` on the pretrained model. Override `inference.N_trajectories_per_task_sample` to something larger than the default (e.g., 8) and `datamodule.batch_size` to something not a multiple of N (e.g., 3) so the interleaving logic is exercised in its less-trivial shape.

### C. Tests

1. **Grammar validity of every generated trajectory.**
   - Read each output parquet.
   - For each `(subject_id, trajectory_idx)` group, recover the token sequence (invert the code → vocab-index mapping).
   - Walk it through `GrammarFSM` and assert every transition is valid.
   - This is the real integration signal.
2. **All N parquets exist and cover all subjects.**
   - Mirrors the existing CLI test but with the larger N to catch "some trajectories missing" bugs under the new interleaving path.
3. **Rolling + grammar correctness.**
   - Run the same fixture with `rolling_generation.max_new_tokens` set large enough to force cross-chunk generation, then re-validate grammar.
4. **Distinctness.**
   - Across trajectories for the same subject, assert non-identical sequences (sampling actually diversifies).
   - Across subjects within one trajectory, assert non-identical sequences (demux actually demuxes).

### D. CLI error-path coverage

Add a few targeted tests that invoke the CLI with bad configs and assert it fails with the expected `ValueError`. These cover error paths in `__main__.py` that currently aren't hit:

- `rolling_generation.max_new_tokens: -1` → validation error.
- `rolling_generation.rolling_context_size: 4` without `max_new_tokens` → validation error.
- `rolling_generation` with an unknown key → validation error.
- Dataset with no `TIMELINE//END` code + rolling requested → error from the auto-eos check.

These are quick subprocess invocations with `pytest.raises` around the check-for-nonzero-exit pattern in `conftest.run_and_check`. Should add a dozen lines or so.

## Dependencies

None. All changes are test-side. No API changes, no new dependencies.

## Where this lives

Keep the in-process `test_pattern_generation.py` as-is — it's a fast, focused unit-style test for `Model.generate` correctness. Add the new CLI-driven tests in a new file `tests/test_pattern_generation_cli.py` (or as a new section of `test_generate_trajectories.py`, TBD). The split keeps the fast grammar test free of subprocess / training-loop overhead while adding the heavier integration coverage.

## Estimated scope

- ~150–250 lines of test code + fixture code.
- One new file for the grammar-MEDS builder, reusing `GrammarFSM`, `PROGRAMS`, `sample_sequence` from the existing pattern test.
- Runtime: target <2 min added to the full test suite (training dominates).

## Non-goals

- Not replacing `test_pattern_generation.py`. The in-process direct-model test stays as the fast feedback loop.
- Not changing any production code paths. This is purely additive test coverage.

**Comment by mmcdermott** [comment]
Delivered by PR #106 (merged into `dev`). The PR body carries `closes #105`; closing manually since dev isn't the default branch. Lives at `tests/grammar/test_cli.py` with 9 tests covering structural correctness, vocab membership, trajectory demux, sampling-grammar signal, 100%-strict greedy grammar validity, and the rolling-generation path.
